### PR TITLE
Issue #2894275 by WidgetsBurritos, vessel_adrift, bighappyface: Clear queue when starting new run

### PR DESCRIPTION
**Pull request by WidgetsBurritos**
Drupal Issue: https://www.drupal.org/node/2894275

This PR clears out the queue before starting a new job. 

Note: I had to do something weird with the integer schema field in regards to the update hook.  Due to a weird bug in Drupal I couldn't use the standard `integer` plugin because it doesn't allow you to set a default value, and it was failing because it was trying to set the value to `NULL` despite being a non-null field type.  To overcome this, I had to override the core integer plugin, with my own version that allows for default values.  

It seems like this would be something core would provide but alas it does not at the present time. I may create a core issue and attempt to resolve this at a later time, but in the mean time, this works.

### CODE REVIEW

**Comment by xinbin** [comment]
+1

File: src/Plugin/Field/FieldType/DefaultIntegerItem.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * )
+ */
+class DefaultIntegerItem extends IntegerItem {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    $schema = parent::schema($field_definition);
+    if (!empty($field_definition->getDefaultValueLiteral()[0]['value'])) {
```
Comment (WidgetsBurritos) [inline src/Plugin/Field/FieldType/DefaultIntegerItem.php]: change to isset()

File: web_page_archive.install [hunk 2]
```diff
@@ -89,3 +91,34 @@ function web_page_archive_schema() {
...
+    ->installFieldStorageDefinition('success_ct', 'web_page_archive_run', 'web_page_archive_run', $storage_definition);
+}
+
+/**
+ * Initialize success_ct values.
+ */
+function web_page_archive_update_8002() {
+  $query = \Drupal::database()->select('web_page_archive_run_revision__field_captures', 'fc');
+  $query->addField('fc', 'revision_id');
+  $query->groupBy("fc.revision_id");
```
Comment (bighappyface) [inline web_page_archive.install]: single quotes

File: web_page_archive.install [hunk 3]
```diff
@@ -89,3 +91,34 @@ function web_page_archive_schema() {
...
+  $query->addExpression("COUNT(*)", 'count');
```
Comment (bighappyface) [inline web_page_archive.install]: single quotes

**Review by bighappyface — APPROVED** [review]
+1 for 06cbea53afbc0259db57a72afafd4025425eefd5

**Pull request merged**